### PR TITLE
Saving Data using Services 

### PR DESCRIPTION
**Pull request by elanghel**

### CODE REVIEW

File: ToDoList/Modules/AddTaskScreen/AddTaskCreenPresenter.swift [hunk 1]
```diff
@@ -7,16 +7,23 @@
...
-    
+protocol AddTaskScreenPresenter {
+    func viewCollectedData(item: Item)
 }
 
-class AddTaskScreenPresenter {
-   
-    weak private var addTaskDelegate: AddTaskScreenDelegate?
+class AddTaskScreenPresenterImp: AddTaskScreenPresenter {
+
```
Comment (vtoloconnicov) [inline ToDoList/Modules/AddTaskScreen/AddTaskCreenPresenter.swift:15]: space

File: ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift [hunk 2]
```diff
@@ -82,6 +85,36 @@ class AddTaskViewController: UIViewController, AddTaskScreenDelegate {
...
+        createTaskButton.setGradientBackground(startColor: UIColor(named: "Dark Purple")!, endColor: UIColor(named: "Light Purple")!)
+        createTaskButton.isUserInteractionEnabled = true
+        let guestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didClickCreateTaskButton))
+        createTaskButton.addGestureRecognizer(guestureRecognizer)
+        
+    }
+    
+    @objc func didClickCreateTaskButton() {
+        
+        let title = taskTextField.text!
```
Comment (vtoloconnicov) [inline ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift:98]: if let text = taskTextField.text {
    title = text 
}
or title = taskTextField.text ?? ""

File: ToDoList/Modules/AddTaskScreen/Model/Item.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//  ToDoList
+//
+//  Created by Elena Anghel on 3/31/21.
+//
+
+import Foundation
+
+struct Item : Codable {
+    
+    var title: String!
```
Comment (vtoloconnicov) [inline ToDoList/Modules/AddTaskScreen/Model/Item.swift:12]: remove the ! from all variables

File: ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mpetrenco) [inline ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist:1]: Can we remove these files from the repo?
These should be filtered by the `.gitignore`

File: ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift [hunk 5]
```diff
@@ -82,6 +85,36 @@ class AddTaskViewController: UIViewController, AddTaskScreenDelegate {
...
+        
+        if title != "" {
```
Comment (vtoloconnicov) [inline ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift:100]: if !title.isEmpty

Comment (mpetrenco) [inline ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift:100]: Use this instead and combine it with the **nil** check mentioned above:
https://developer.apple.com/documentation/swift/string/2946268-isempty

File: ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist (hunk 4 above)
Comment (vtoloconnicov) [reply ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist:1]: it's in gitignore. Elena just didn't have it updated at first commit

File: ToDoList/Modules/AddTaskScreen/Model/Item.swift (hunk 3 above)
Comment (elanghel) [reply ToDoList/Modules/AddTaskScreen/Model/Item.swift:12]: done

File: ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist (hunk 4 above)
Comment (elanghel) [reply ToDoList.xcodeproj/project.xcworkspace/xcuserdata/elanghel.xcuserdatad/IDEFindNavigatorScopes.plist:1]: I will be careful not to appear in other projects  in the future

File: ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift [hunk 6]
```diff
@@ -82,6 +85,41 @@ class AddTaskViewController: UIViewController, AddTaskScreenDelegate {
...
+        createTaskButton.setGradientBackground(startColor: UIColor(named: "Dark Purple")!, endColor: UIColor(named: "Light Purple")!)
+        createTaskButton.isUserInteractionEnabled = true
+        let guestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didClickCreateTaskButton))
+        createTaskButton.addGestureRecognizer(guestureRecognizer)
+        
+    }
+    
+    @objc func didClickCreateTaskButton() {
+        
+        let horseName = "Silver"
```
Comment (vtoloconnicov) [inline ToDoList/Modules/AddTaskScreen/Views/AddTaskViewController.swift:98]: new feature?

**Pull request merged**